src/ir/field_selector.h

```diff
@@ -0,0 +1,50 @@
+#ifndef SRC_IR_FIELD_SELECTOR_H_
+#define SRC_IR_FIELD_SELECTOR_H_
+
+#include <string>
+
+#include "absl/strings/str_cat.h"
+
+namespace raksha::ir {
+
+// Represents descending through the type tree by taking the field of an
+// EntityType in an AccessPath.
+class FieldSelector {
+ public:
+  FieldSelector(std::string field_name) : field_name_(std::move(field_name)) {}
+
+  // Allow moving the FieldSelector.
+  FieldSelector(FieldSelector &&other) = default;
+  FieldSelector &operator=(FieldSelector &&other) = default;
+
+  // Prints the string representing dereferencing the field in the AccessPath.
+  // This will just be the "." punctuation plus the name of the field.
+  std::string ToString() const {
+    return absl::StrCat(".", field_name_);
+  }
```

Comment:
Single line for better readability?

```suggestion
  std::string ToString() const { return absl::StrCat(".", field_name_); }
```

Reply:
Done.